### ISSUE TITLE
telegram-desktop becomes very slow after opening an image

### ISSUE THREAD

**Issue by VitalyAnkh** [telegramdesktop/tdesktop#17121]
### Steps to reproduce

1. Open an image in the message, then close the image window
2. The telegram-desktop becomes very slow/lagging

### Expected behaviour

No lagging after opening an image

### Actual behaviour

lagging after opening an image

### Operating system

Arch Linux, rolling, Linux 5.14.12-zen1-1-zen x86_64, Gnome 41

### Version of Telegram Desktop

3.1.9

### Installation source

Other (unofficial) source

### Logs

```shell
[2021.10.19 00:04:11] Launched version: 3001009, install beta: [FALSE], alpha: 0, debug mode: [FALSE]
[2021.10.19 00:04:11] Executable dir: /usr/bin/, name: telegram-desktop
[2021.10.19 00:04:11] Initial working dir: /home/vitalyr/
[2021.10.19 00:04:11] Working dir: /home/vitalyr/.local/share/TelegramDesktop/
[2021.10.19 00:04:11] Command line: telegram-desktop --
[2021.10.19 00:04:11] Executable path before check: /usr/bin/telegram-desktop
[2021.10.19 00:04:11] Logs started
[2021.10.19 00:04:11] Launcher filename: telegramdesktop.desktop
[2021.10.19 00:04:11] IBus portal is present! Using it.
[2021.10.19 00:04:11] Connecting local socket to /run/user/1000/1fcd10e2c7e1111f370d45725973bf29-{87A94AB0-E370-4cde-98D3-ACC110C5967D}...
[2021.10.19 00:04:11] This is the only instance of Telegram, starting server and app...
[2021.10.19 00:04:11] Moved logging from '/home/vitalyr/.local/share/TelegramDesktop/log_start0.txt' to '/home/vitalyr/.local/share/TelegramDesktop/log.txt'!
[2021.10.19 00:04:11] Primary screen DPI: 96
[2021.10.19 00:04:11] Found non-trivial Device Pixel Ratio: 2
[2021.10.19 00:04:11] Environmental variables: QT_DEVICE_PIXEL_RATIO=''
[2021.10.19 00:04:11] Environmental variables: QT_SCALE_FACTOR=''
[2021.10.19 00:04:11] Environmental variables: QT_AUTO_SCREEN_SCALE_FACTOR='0'
[2021.10.19 00:04:11] Environmental variables: QT_SCREEN_SCALE_FACTORS=''
[2021.10.19 00:04:11] Icon theme: WhiteSur
[2021.10.19 00:04:11] Fallback icon theme: 
[2021.10.19 00:04:11] App Info: reading settings...
[2021.10.19 00:04:11] App Info: reading encrypted settings...
[2021.10.19 00:04:11] Lang Info: Loaded cached, keys: 3424
[2021.10.19 00:04:11] OpenAL Logging Level: (not set)
[2021.10.19 00:04:11] Audio Playback Devices: 方糖(B5 C0);EasyEffects Sink;GP106 High Definition Audio Controller 数字立体声(HDMI);内置音频 模拟立体声
[2021.10.19 00:04:11] Audio Playback Default Device: 方糖(B5 C0)
[2021.10.19 00:04:11] Audio Capture Devices: 内置音频 模拟立体声;Monitor of 方糖(B5 C0);Monitor of EasyEffects Sink;EasyEffects Source;Monitor of GP106 High Definition Audio Controller 数字立体声(HDMI);Monitor of 内置音频 模拟立体声
[2021.10.19 00:04:11] Audio Capture Default Device: 内置音频 模拟立体声
[2021.10.19 00:04:11] Not using D-Bus global menu.
[2021.10.19 00:04:11] Not using Unity launcher counter.
[2021.10.19 00:04:11] System tray available: [TRUE]
[2021.10.19 00:04:11] App Info: reading accounts info...
[2021.10.19 00:04:11] App Info: reading encrypted info...
[2021.10.19 00:04:11] App Info: reading map...
[2021.10.19 00:04:11] App Info: reading encrypted map...
[2021.10.19 00:04:11] App Info: reading encrypted user settings...
[2021.10.19 00:04:11] App Info: encrypted user settings read.
[2021.10.19 00:04:11] App Info: reading encrypted mtp data...
[2021.10.19 00:04:11] MTP Info: read keys, current: 5, to destroy: 0
[2021.10.19 00:04:11] Map read time: 11
[2021.10.19 00:04:11] App Info: reading encrypted mtp config...
[2021.10.19 00:04:11] HTTP Error: host not found - Host 2001:b28:f23f:f005::a not found
[2021.10.19 00:04:11] App Info: reading map...
[2021.10.19 00:04:11] App Info: reading encrypted map...
[2021.10.19 00:04:11] App Info: reading encrypted user settings...
[2021.10.19 00:04:11] App Info: encrypted user settings read.
[2021.10.19 00:04:11] App Info: reading encrypted mtp data...
[2021.10.19 00:04:11] MTP Info: read keys, current: 5, to destroy: 0
[2021.10.19 00:04:11] Map read time: 1
[2021.10.19 00:04:11] App Info: reading encrypted mtp config...
[2021.10.19 00:04:11] HTTP Error: host not found - Host 2001:b28:f23d:f001::a not found
[2021.10.19 00:04:11] Export Info: Destroy top bar by controller removal.
[2021.10.19 00:04:11] OpenGL Profile: Compatibility.
[2021.10.19 00:04:11] OpenGL Renderer: NVIDIA GeForce GTX 1060/PCIe/SSE2
[2021.10.19 00:04:11] OpenGL Vendor: NVIDIA Corporation
[2021.10.19 00:04:11] OpenGL Version: 4.6.0 NVIDIA 495.29.05
[2021.10.19 00:04:11] OpenGL Extensions: GL_EXT_texture_compression_latc, GL_NV_ES3_1_compatibility, GL_ARB_shader_storage_buffer_object, GL_ARB_vertex_buffer_object, GL_ARB_fragment_shader, GL_EXT_point_parameters, GL_EXT_multiview_texture_multisample, GL_EXT_blend_subtract, GL_ARB_texture_query_lod, GL_ARB_framebuffer_no_attachments, GL_KHR_robustness, GL_ARB_texture_rectangle, GL_ARB_program_interface_query, GL_ARB_texture_border_clamp, GL_S3_s3tc, GL_ARB_sync, GL_NV_conservative_raster, GL_NV_fragment_program_option, GL_NV_texture_rectangle_compressed, GL_ARB_shader_subroutine, GL_ARB_debug_output, GL_ARB_draw_elements_base_vertex, GL_NV_primitive_restart, GL_NV_gpu_shader5, GL_NV_sample_mask_override_coverage, GL_EXT_separate_specular_color, GL_ARB_texture_gather, GL_ARB_texture_compression_bptc, GL_ARB_vertex_program, GL_NV_bindless_multi_draw_indirect_count, GL_NV_shader_thread_group, GL_ARB_texture_stencil8, GL_NV_shader_storage_buffer_object, GL_EXT_texture_lod, GL_ARB_texture_compression_rgtc, GL_NV_texture_multisample, GL_EXT_texture_env_add, GL_EXT_texture3D, GL_NV_compute_program5, GL_ARB_transpose_matrix, GL_ARB_vertex_array_bgra, GL_ARB_texture_view, GL_EXT_draw_range_elements, GL_NV_fog_distance, GL_SGIX_shadow, GL_NV_blend_equation_advanced, GL_ARB_clear_texture, GL_ARB_explicit_attrib_location, GL_ARB_ES3_1_compatibility, GL_ARB_stencil_texturing, GL_NV_path_rendering_shared_edge, GL_SGIS_texture_lod, GL_NV_vertex_program2_option, GL_ARB_gpu_shader_fp64, GL_NV_geometry_shader4, GL_NV_memory_attachment, GL_NV_gpu_program5_mem_extended, GL_ARB_shading_language_420pack, GL_ATI_texture_mirror_once, GL_EXTX_framebuffer_mixed_formats, GL_NV_stereo_view_rendering, GL_NVX_conditional_render, GL_AMD_multi_draw_indirect, GL_NV_copy_image, GL_NV_vertex_program, GL_ARB_texture_mirrored_repeat, GL_EXT_texture_env_combine, GL_NV_timeline_semaphore, GL_EXT_stencil_two_side, GL_ARB_draw_instanced, GL_NV_shader_atomic_int64, GL_ARB_shader_group_vote, GL_ARB_ES2_compatibility, GL_ARB_transform_feedback_instanced, GL_EXT_raster_multisample, GL_EXT_framebuffer_multisample, GL_ARB_window_pos, GL_ARB_uniform_buffer_object, GL_NV_framebuffer_multisample_coverage, GL_ARB_conservative_depth, GL_NV_light_max_exponent, GL_NV_command_list, GL_EXT_texture_swizzle, GL_NV_depth_buffer_float, GL_EXT_framebuffer_sRGB, GL_NV_feature_query, GL_NV_half_float, GL_NV_geometry_shader_passthrough, GL_ARB_ES3_compatibility, GL_NV_vertex_attrib_integer_64bit, GL_EXT_texture_filter_minmax, GL_ARB_internalformat_query, GL_EXT_bgra, GL_NV_uniform_buffer_unified_memory, GL_ARB_map_buffer_range, GL_ARB_shading_language_include, GL_ARB_shader_draw_parameters, GL_ARB_shading_language_100, GL_EXT_texture_storage, GL_ARB_internalformat_query2, GL_EXT_texture_filter_anisotropic, GL_EXT_draw_buffers2, GL_ARB_vertex_attrib_64bit, GL_NV_framebuffer_mixed_samples, GL_NV_texture_shader3, GL_NV_bindless_texture, GL_EXT_shadow_funcs, GL_ARB_shader_atomic_counters, GL_ARB_enhanced_layouts, GL_ARB_blend_func_extended, GL_ARB_texture_compression, GL_ARB_direct_state_access, GL_EXT_texture_compression_s3tc, GL_KHR_robust_buffer_access_behavior, GL_NV_texture_shader2, GL_NV_query_resource_tag, GL_EXT_texture_shadow_lod, GL_ARB_shader_texture_lod, GL_EXT_blend_func_separate, GL_EXT_abgr, GL_EXT_texture_env_dot3, GL_NV_float_buffer, GL_ARB_sample_shading, GL_NV_shader_atomic_counters, GL_AMD_vertex_shader_layer, GL_ARB_clear_buffer_object, GL_ARB_sample_locations, GL_EXT_bindable_uniform, GL_NV_shader_thread_shuffle, GL_ARB_draw_buffers_blend, GL_EXT_sparse_texture2, GL_IBM_texture_mirrored_repeat, GL_ARB_pipeline_statistics_query, GL_EXT_texture_mirror_clamp, GL_ARB_parallel_shader_compile, GL_ARB_point_parameters, GL_NV_blend_square, GL_NVX_progress_fence, GL_EXT_texture_sRGB, GL_ARB_shader_viewport_layer_array, GL_ARB_occlusion_query, GL_NV_clip_space_w_scaling, GL_NV_draw_texture, GL_ARB_tessellation_shader, GL_ARB_texture_buffer_object, GL_ARB_texture_env_dot3, GL_ARB_fragment_layer_viewport, GL_NV_path_rendering, GL_KHR_parallel_shader_compile, GL_ARB_sparse_buffer, GL_ARB_spirv_extensions, GL_ARB_texture_cube_map_array, GL_EXT_stencil_wrap, GL_ARB_seamless_cube_map, GL_ARB_texture_non_power_of_two, GL_ARB_compute_variable_group_size, GL_ARB_texture_env_add, GL_NV_packed_depth_stencil, GL_EXT_pixel_buffer_object, GL_ARB_pixel_buffer_object, GL_NVX_blend_equation_advanced_multi_draw_buffers, GL_ARB_texture_filter_minmax, GL_EXT_texture_array, GL_NV_explicit_multisample, GL_EXT_provoking_vertex, GL_ARB_vertex_array_object, GL_NV_multisample_filter_hint, GL_ARB_half_float_pixel, GL_NV_blend_equation_advanced_coherent, GL_ARB_texture_multisample, GL_ARB_texture_buffer_range, GL_ARB_multisample, GL_ARB_geometry_shader4, GL_NV_fragment_shader_interlock, GL_NV_gpu_program5, GL_NVX_gpu_memory_info, GL_ARB_compute_shader, GL_ARB_sparse_texture2, GL_ARB_invalidate_subdata, GL_EXT_vertex_attrib_64bit, GL_EXT_semaphore_fd, GL_ARB_shader_image_size, GL_NV_shader_atomic_fp16_vector, GL_ARB_sampler_objects, GL_NV_register_combiners2, GL_ARB_texture_mirror_clamp_to_edge, GL_ARB_fragment_shader_interlock, GL_ARB_bindless_texture, GL_EXT_shader_image_load_formatted, GL_NV_gpu_program4, GL_EXT_shader_integer_mix, GL_NV_shader_subgroup_partitioned, GL_EXT_post_depth_coverage, GL_NV_internalformat_sample_query, GL_EXT_polygon_offset_clamp, GL_NV_depth_clamp, GL_EXT_transform_feedback2, GL_EXT_framebuffer_blit, GL_ARB_shader_image_load_store, GL_ARB_draw_buffers, GL_NV_gpu_multicast, GL_EXT_shader_image_load_store, GL_ARB_vertex_attrib_binding, GL_ARB_depth_buffer_float, GL_ARB_compressed_texture_pixel_storage, GL_ARB_shader_objects, GL_ARB_vertex_shader, GL_ARB_robust_buffer_access_behavior, GL_NV_texture_compression_vtc, GL_ARB_shader_bit_encoding, GL_EXT_texture_compression_rgtc, GL_AMD_seamless_cubemap_per_texture, GL_ARB_compatibility, GL_ARB_shadow, GL_ARB_texture_env_crossbar, GL_ARB_timer_query, GL_ARB_texture_storage_multisample, GL_NV_gpu_program_fp64, GL_EXT_direct_state_access, GL_ARB_query_buffer_object, GL_ARB_shader_texture_image_samples, GL_EXT_draw_instanced, GL_EXT_framebuffer_object, GL_NV_robustness_video_memory_purge, GL_ARB_cull_distance, GL_NV_texture_barrier, GL_ARB_texture_rg, GL_ARB_transform_feedback_overflow_query, GL_ARB_conditional_render_inverted, GL_NV_point_sprite, GL_NV_shader_atomic_float, GL_NV_vertex_buffer_unified_memory, GL_NV_fragment_program2, GL_ARB_half_float_vertex, GL_EXT_depth_bounds_test, GL_NV_draw_vulkan_image, GL_ARB_clip_control, GL_EXT_timer_query, GL_ARB_provoking_vertex, GL_KHR_shader_subgroup, GL_EXT_texture_sRGB_R8, GL_ARB_indirect_parameters, GL_NV_occlusion_query, GL_ARB_base_instance, GL_ARB_occlusion_query2, GL_EXT_gpu_program_parameters, GL_EXT_window_rectangles, GL_EXT_texture_buffer_object, GL_ARB_texture_barrier, GL_NV_shader_atomic_float64, GL_KHR_blend_equation_advanced, GL_ARB_sparse_texture, GL_ARB_gl_spirv, GL_ARB_vertex_type_10f_11f_11f_rev, GL_ARB_shader_precision, GL_IBM_rasterpos_clip, GL_EXT_vertex_array_bgra, GL_ARB_texture_float, GL_ARB_sparse_texture_clamp, GL_KTX_buffer_region, GL_EXT_geometry_shader4, GL_ARB_post_depth_coverage, GL_NV_fragment_program, GL_EXT_blend_minmax, GL_EXT_texture_edge_clamp, GL_EXT_texture_integer, GL_ARB_copy_image, GL_ARB_shading_language_packing, GL_EXT_texture_object, GL_EXT_multiview_timer_query, GL_NV_query_resource, GL_ARB_multi_bind, GL_ARB_explicit_uniform_location, GL_EXT_secondary_color, GL_KHR_context_flush_control, GL_NV_blend_minmax_factor, GL_NV_vertex_program3, GL_SGIS_generate_mipmap, GL_NV_fill_rectangle, GL_NV_transform_feedback2, GL_OVR_multiview, GL_ARB_multi_draw_indirect, GL_ARB_fragment_coord_conventions, GL_NV_texture_rectangle, GL_ARB_copy_buffer, GL_SGIX_depth_texture, GL_EXT_packed_float, GL_ARB_transform_feedback2, GL_EXT_vertex_array, GL_ARB_seamless_cubemap_per_texture, GL_ARB_depth_texture, GL_ATI_texture_float, GL_ARB_texture_storage, GL_NV_ES1_1_compatibility, GL_NV_fence, GL_EXT_blend_equation_separate, GL_EXT_separate_shader_objects, GL_ARB_draw_indirect, GL_ARB_framebuffer_sRGB, GL_EXT_framebuffer_multisample_blit_scaled, GL_ARB_imaging, GL_NV_alpha_to_coverage_dither_control, GL_ARB_gpu_shader5, GL_NV_parameter_buffer_object2, GL_NV_shader_buffer_load, GL_ARB_gpu_shader_int64, GL_EXT_texture_cube_map, GL_NV_copy_depth_to_color, GL_EXT_multi_draw_arrays, GL_ARB_multitexture, GL_ARB_ES3_2_compatibility, GL_NV_vertex_program2, GL_ARB_derivative_control, GL_EXT_gpu_shader4, GL_NV_parameter_buffer_object, GL_EXT_texture_sRGB_decode, GL_ARB_shader_clock, GL_EXT_texture_shared_exponent, GL_EXT_fog_coord, GL_NV_multisample_coverage, GL_ARB_texture_query_levels, GL_ARB_robustness, GL_ARB_fragment_program, GL_EXT_blend_color, GL_NV_register_combiners, GL_ARB_texture_env_combine, GL_OES_EGL_image, GL_NV_gpu_program4_1, GL_KHR_debug, GL_EXT_memory_object_fd, GL_NV_texture_env_combine4, GL_NV_transform_feedback, GL_EXT_memory_object, GL_ARB_fragment_program_shadow, GL_NV_conservative_raster_pre_snap_triangles, GL_ARB_map_buffer_alignment, GL_ARB_shader_atomic_counter_ops, GL_ARB_texture_cube_map, GL_ARB_texture_filter_anisotropic, GL_EXT_packed_pixels, GL_KHR_blend_equation_advanced_coherent, GL_ARB_texture_swizzle, GL_ARB_transform_feedback3, GL_ARB_buffer_storage, GL_AMD_vertex_shader_viewport_index, GL_NV_memory_object_sparse, GL_ARB_separate_shader_objects, GL_ARB_vertex_type_2_10_10_10_rev, GL_ARB_texture_rgb10_a2ui, GL_NV_vertex_program1_1, GL_ARB_instanced_arrays, GL_ARB_color_buffer_float, GL_EXT_Cg_shader, GL_NV_bindless_multi_draw_indirect, GL_EXT_packed_depth_stencil, GL_ARB_texture_buffer_object_rgb32, GL_SUN_slice_accum, GL_NV_conservative_raster_dilate, GL_EXT_compiled_vertex_array, GL_EXT_rescale_normal, GL_OVR_multiview2, GL_ARB_depth_clamp, GL_ARB_get_texture_sub_image, GL_ARB_polygon_offset_clamp, GL_NV_texgen_reflection, GL_NV_fragment_coverage_to_color, GL_ARB_shader_ballot, GL_ARB_viewport_array, GL_ARB_get_program_binary, GL_KHR_no_error, GL_ARB_arrays_of_arrays, GL_NV_conditional_render, GL_ATI_draw_buffers, GL_NV_viewport_swizzle, GL_EXT_semaphore, GL_NV_viewport_array2, GL_EXT_import_sync_object, GL_EXT_texture_lod_bias, GL_ARB_point_sprite, GL_NV_sample_locations, GL_NV_texture_shader, GL_ARB_framebuffer_object, GL_EXT_EGL_image_storage, GL_EXT_texture_compression_dxt1
[2021.10.19 00:04:11] OpenGL: QOpenGLContext created, version: 4.4.
[2021.10.19 00:04:11] OpenGL: [TRUE] (OverlayWidget)
[2021.10.19 00:04:11] Using SNI tray icon.
[2021.10.19 00:04:12] Notification daemon product name: gnome-shell
[2021.10.19 00:04:12] Notification daemon vendor name: GNOME
[2021.10.19 00:04:12] Notification daemon version: 41.0
[2021.10.19 00:04:12] Notification daemon specification version: 1.2
[2021.10.19 00:04:12] Notification daemon capabilities: actions, body, body-markup, icon-static, persistence, sound
[2021.10.19 00:04:12] API Warning: not loaded minimal channel applied.
[2021.10.19 00:04:13] HTTP Error: host not found - Host 2001:b28:f23d:f001::a not found
[2021.10.19 00:04:13] HTTP Error: host not found - Host 2001:b28:f23f:f005::a not found
[2021.10.19 00:04:14] API Warning: not loaded minimal channel applied.
[2021.10.19 00:04:14] API Warning: not loaded minimal channel applied.
[2021.10.19 00:04:14] API Warning: not loaded minimal channel applied.
[2021.10.19 00:04:25] API Warning: not loaded minimal channel applied.
[2021.10.19 00:04:34] HTTP Error: host not found - Host 2001:b28:f23d:f001::a not found
[2021.10.19 00:04:52] Message Info: bad message notification received (error_code 33) for msg_id = 7020437975713752428, seq_no = 318
[2021.10.19 00:04:52] Message Info: bad message notification received, msgId 7020437975713752428, error_code 33
[2021.10.19 00:04:52] HTTP Error: host not found - Host 2001:b28:f23f:f005::a not found
[2021.10.19 00:05:26] API Warning: not loaded minimal channel applied.
[2021.10.19 00:05:26] HTTP Error: host not found - Host 2001:b28:f23f:f005::a not found
[2021.10.19 00:05:43] HTTP Error: host not found - Host 2001:67c:4e8:f004::a not found
[2021.10.19 00:05:55] HTTP Error: host not found - Host 2001:b28:f23f:f005::a not found
[2021.10.19 00:05:55] HTTP Error: host not found - Host 2001:67c:4e8:f004::b not found
[2021.10.19 00:06:40] Skipping message, because it is already in blocks!
[2021.10.19 00:06:53] HTTP Error: host not found - Host 2001:b28:f23d:f001::a not found
[2021.10.19 00:06:59] HTTP Error: host not found - Host 2001:b28:f23f:f005::a not found
[2021.10.19 00:07:17] API Warning: not loaded minimal channel applied.
[2021.10.19 00:07:28] HTTP Error: host not found - Host 2001:b28:f23f:f005::a not found
[2021.10.19 00:08:09] HTTP Error: host not found - Host 2001:b28:f23d:f001::a not found
[2021.10.19 00:08:24] HTTP Error: host not found - Host 2001:b28:f23f:f005::a not found
[2021.10.19 00:09:01] HTTP Error: host not found - Host 2001:b28:f23d:f001::a not found
[2021.10.19 00:09:03] HTTP Error: host not found - Host 2001:b28:f23f:f005::a not found
[2021.10.19 00:09:39] RPC Error: request 388 got fail with code 400, error CHAT_ADMIN_REQUIRED
```

**Comment by VitalyAnkh** [comment]
It seems to be fixed by version 3.1.11.

**Comment by karanssh** [comment]
can confirm, update to latest has fixed this on windows 10. This bug can be closed
